### DISCUSSION TITLE
Link to pdf returns a 404

### DISCUSSION THREAD

**Discussion by Anaxarchos** [pimcore/pimcore#13497] [Q&A / Support] [answered]
Hi,
when I link to a pdf that I did upload to a folder in the assets directory and that I can preview in the backend, it returns a 404 error.
But when I link to a pdf that was uploaded via ssh directly to the public folder everything works fine. 
The directory permissions in the public directory are 0750 for directories and 0640 for all files and they all belong to the same owner and the same group.
And images are all displayed as expected. I know they live in the public/var/tmp/thumbnails directory and not in public/var/assets.
Is the public/var/assets directory blocked by pimcore? I did not find anything about that in the documentation.
Any help is appreciated. Thank you.

**Comment by mattamon** [comment, marked as answer]
Hey @Anaxarchos,

what server are you using? Could it be a .htaccess Problem?
If you upload the asset via ssh does the owner match the rest?
Like pimcore states in the docs all assets are public -> https://pimcore.com/docs/pimcore/current/Development_Documentation/Assets/Restricting_Public_Asset_Access.html

**Reply by Anaxarchos** [reply]
Hi @mattamon,
I use nginx, so no .htaccess is involved.
I can upload a pdf file. I can open it in the backend (the deeplink is working), but when I link to it, I get a 404 error. 
And when I upload a jpg or png to the same directory I can access it via its link.

**Reply by mattamon** [reply]
Hey @Anaxarchos,

is really sounds like a NGINX config problem. 

If I understand it correctly:
http://localhost/test/1.pdf -> does not work
http://localhost/test/1.jpg -> does work

Maybe you could share parts of your NGINX Config?

**Reply by Anaxarchos** [reply]
Hi @mattamon,
after playing around a bit I found out, that when I add 'var/assets' to the url, the pdf shows up as expected.
Now in the article below the documentation states, that one can override the File Storage Setup.
https://pimcore.com/docs/pimcore/10.2/Development_Documentation/Installation_and_Upgrade/System_Setup_and_Hosting/File_Storage_Setup.html

I did not find this config file in my pimcore application. And as far as I can see, the default values should already link to the desired output and it is working for images which are located in '/public/var/tmp/thumbnails'

**Reply by mattamon** [reply]
Hey @Anaxarchos,

could you please provide an answer to my question above?


Like I said I think it is a NGINX config problem. 
Without some details about your configuration, I cannot dive deeper into the problem. 

You can find a full config here: https://pimcore.com/docs/pimcore/current/Development_Documentation/Installation_and_Upgrade/System_Setup_and_Hosting/Nginx_Configuration.html

Can you please check the root and the asset location?
```yaml
root /var/www/pimcore/public;
```

```yaml
  # Assets
  # Still use a whitelist approach to prevent each and every missing asset to go through the PHP Engine.
  location ~* ^(?!/admin)(.+?)\.((?:css|js)(?:\.map)?|jpe?g|gif|png|svgz?|eps|exe|gz|zip|mp\d|m4a|ogg|ogv|webm|pdf|docx?|xlsx?|pptx?)$ {
      try_files /var/assets$uri $uri =404;
      expires 2w;
      access_log off;
      log_not_found off;
      add_header Cache-Control "public";
  }
```

**Reply by Anaxarchos** [reply]
I had this directive in my nginx config, but then I cannot login to the backend anymore, because the router.js throws a 500 Error without any further information. And I thought that the (?!/admin) expression indicates that this directive should not bother any admin route. But my regex knowledge is quite poor.
